rrule/src/parser/regex.rs

```diff
@@ -0,0 +1,261 @@
+//! Utility functions around the regexes we use for parsing rrule strings.
+use std::str::FromStr;
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use super::{content_line::PropertyName, ParseError};
+
+lazy_static! {
+    static ref DATESTR_RE: Regex =
+        Regex::new(r"(?m)^(\d{4})(\d{2})(\d{2})(T(\d{2})(\d{2})(\d{2})(Z?))?$").unwrap();
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct ParsedDateString {
+    pub year: i32,
+    pub month: u32,
+    pub day: u32,
+    pub time: Option<ParsedDateStringTime>,
+    pub flags: ParsedDateStringFlags,
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct ParsedDateStringFlags {
+    pub zulu_timezone_set: bool,
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct ParsedDateStringTime {
+    pub hour: u32,
+    pub min: u32,
+    pub sec: u32,
+}
+
+/// Parses a date string with format `YYYYMMDD(THHMMSSZ)` where the part in paranthesis
+/// is optional. It returns [`ParsedDateString`].
+pub(crate) fn parse_datestring(val: &str) -> Result<ParsedDateString, ()> {
+    let captures = DATESTR_RE.captures(val).ok_or(())?;
+
+    let year = captures
+        .get(1)
+        .ok_or(())?
+        .as_str()
+        .parse()
+        .map_err(|_| ())?;
```

Comment:
I don't like returning an empty tuple as error, maybe a new error type, but not an empty tuple.

If you don't want to repeat the error type, you can have a code like this:
```
captures
        .get(1)
        .and_then(|v| v.as_str().parse().ok())
        .ok_or(())?;
```